noir-projects/aztec-nr/aztec/src/prelude.nr

```diff
@@ -0,0 +1,27 @@
+use dep::protocol_types::{
+    address::{
+        AztecAddress, 
+        EthAddress,
+    },
+};
+use crate::{
+    state_vars::{
+        map::Map, 
+        public_state::PublicState, 
+        singleton::Singleton,
+        immutable_singleton::ImmutableSingleton,
+        set::Set,
+        stable_public_state::StablePublicState,
+    },
+    log::{
+        emit_unencrypted_log,
+        emit_encrypted_log,
+    },
+    context::PrivateContext,
+    note::{
+        note_header::NoteHeader,
+        note_interface::NoteInterface,
+        note_getter_options::NoteGetterOptions, 
```

Comment:
these imports don't seem to be removed from the contracts?

Reply:
Will remove!

Reply:
I was thinking we could merge this as is and in #4610 we inject `compute_note_hash_and_nullifier` as to keep separation with the goal of the PRs. We can then have a workflow that any future macros which require certain names from `aztec` must also include it in the prelude.

Reply:
Actually I guess we wouldn't be removing any imports from `noir-contracts` by adding `compute_note_hash_and_nullifier` here so I am fine adding it here as well. I just figured because this PR should go in before #4610 anyway that it would be fine for the extra import to be injected as part of #4610.

Reply:
Agreed, makes perfect sense.